console/src/view/resource.rs

```diff
@@ -0,0 +1,150 @@
+    },
+};
+use std::{cell::RefCell, rc::Rc};
+use tui::{
+    layout::{self, Layout},
+    text::{Span, Spans, Text},
+    widgets::{Block, Paragraph},
+};
+
+pub(crate) struct ResourceView {
+    resource: Rc<RefCell<Resource>>,
+    async_ops_table: TableListState<AsyncOpsTable>,
+    initial_render: bool,
+}
+
+impl ResourceView {
+    pub(super) fn new(resource: Rc<RefCell<Resource>>) -> Self {
+        ResourceView {
+            resource,
+            async_ops_table: TableListState::<AsyncOpsTable>::default(),
+            initial_render: true,
+        }
+    }
+
+    pub(crate) fn update_input(&mut self, event: input::Event) {
+        self.async_ops_table.update_input(event)
+    }
+
+    pub(crate) fn render<B: tui::backend::Backend>(
+        &mut self,
+        styles: &view::Styles,
+        frame: &mut tui::terminal::Frame<B>,
+        area: layout::Rect,
+        state: &mut State,
+    ) {
+        let resource = &*self.resource.borrow();
+
+        let (controls_area, stats_area, async_ops_area) = {
+            let chunks = Layout::default()
+                .direction(layout::Direction::Vertical)
+                .constraints(
+                    [
+                        // controls
+                        layout::Constraint::Length(1),
+                        // resource stats
+                        layout::Constraint::Length(8),
+                        // async ops
+                        layout::Constraint::Percentage(60),
+                    ]
+                    .as_ref(),
+                )
+                .split(area);
+            (chunks[0], chunks[1], chunks[2])
+        };
+
+        let stats_area = Layout::default()
+            .direction(layout::Direction::Horizontal)
+            .constraints(
+                [
+                    layout::Constraint::Percentage(50),
+                    layout::Constraint::Percentage(50),
+                ]
+                .as_ref(),
+            )
+            .split(stats_area);
+
+        let controls = Spans::from(vec![
+            Span::raw("controls: "),
+            bold(styles.if_utf8("\u{238B} esc", "esc")),
+            Span::raw(" = return to task list, "),
+            bold("q"),
+            Span::raw(" = quit"),
+        ]);
+
+        let mut overview = Vec::with_capacity(6);
+        overview.push(Spans::from(vec![
+            bold("ID: "),
+            Span::raw(format!("{} ", resource.id())),
+        ]));
+
+        overview.push(Spans::from(vec![
+            bold("Parent ID: "),
+            Span::raw(match resource.parent_id() {
+                Some(id) => state
+                    .resources_state()
+                    .resource(id)
+                    .and_then(|r| r.upgrade())
+                    .map(|r| {
+                        let r = r.borrow();
+                        format!("{} ({}::{})", r.id(), r.target(), r.concrete_type())
+                    })
+                    .unwrap_or(format!("{}", id)),
```

Comment:
ids, targets, and concrete types aren't going to change, so it would be nice to be able to avoid formatting these strings every time.